src/coreclr/dlls/mscordbi/CMakeLists.txt

```
@@ -100,7 +100,10 @@ elseif(CLR_CMAKE_HOST_UNIX)
         mscordaccore
     )
 
```

```suggestion

    # Before llvm 16, lld was setting `--undefined-version` by default. The default was
    # flipped to `--no-undefined-version` in lld 16, so we will explicitly set it to
    # `--undefined-version` for our use-case.
    #
    include(CheckLinkerFlag OPTIONAL)
    if(COMMAND check_linker_flag)
        check_linker_flag(CXX -Wl,--undefined-version LINKER_SUPPORTS_UNDEFINED_VERSION)
        if (LINKER_SUPPORTS_UNDEFINED_VERSION)
            add_linker_flag(-Wl,--undefined-version)
        endif(LINKER_SUPPORTS_UNDEFINED_VERSION)
    endif(COMMAND check_linker_flag)

```

Fixes the compilation error and `ldd -r libmscordbi.so` reports no missing symbols.